engine/src/test/scala/org/scalarules/dsl/nl/grammar/AfrondingsWordsTest.scala

```diff
@@ -0,0 +1,226 @@
+package org.scalarules.dsl.nl.grammar
+
+import org.scalacheck.Gen
+import org.scalacheck.Prop.{all, forAll}
+import org.scalarules.dsl.nl.grammar.AfrondingsTestBerekeningGlossary._
+import org.scalarules.engine._
+import org.scalarules.finance.nl._
+import org.scalarules.utils.InternalTestUtils.runAndExtractFact
+import org.scalatest.PropSpec
+import org.scalatest.prop.Checkers
+
+import scala.language.postfixOps
+
+class AfrondingsWordsTest extends PropSpec with Checkers {
+
+  property("halfNaarEven should match BigDecimal.RoundingMode.HALF_EVEN for positive doubles") {
+    check(
+      forAll (for {testGetal <- Gen.posNum[Double]} yield testGetal) { testGetal: Double =>
+
+        val context: Context = Map(startPercentage -> BigDecimal(testGetal).procent,
+                                    afrondingsType -> "halfNaarEven")
+
+        val uitkomst: Option[Percentage] = runAndExtractFact(context, new AfrondingsTestBerekening().berekeningen, afgerondPercentageHalfEven)
```

Comment:
If all derivations are run all the time, you can just `run` the engine here, and check every value returned in the context against its corresponding rounding mode.

After doing so, you might want to consider upping the number of checks performed on this property.